drivers/storage/portworx/util/csi_generator.go

```diff
@@ -90,28 +93,20 @@ func (g *CSIGenerator) GetCSIConfiguration() *CSIConfiguration {
 	k8sVer1_14, _ := version.NewVersion("1.14")
 	k8sVer1_16, _ := version.NewVersion("1.16")
 	k8sVer1_17, _ := version.NewVersion("1.17")
-	pxVer2_1, _ := version.NewVersion("2.1")
 	pxVer2_2, _ := version.NewVersion("2.2")
 
 	var cv *CSIConfiguration
-	if g.pxVersion.GreaterThan(pxVer2_1) || g.pxVersion.Equal(pxVer2_1) {
-		// Portworx version >= 2.1
-		// Can only use Csi 1.0 in Kubernetes 1.13+
-		if g.kubeVersion.GreaterThan(k8sVer1_13) || g.kubeVersion.Equal(k8sVer1_13) {
-			cv = g.setSidecarContainerVersionsV1_0()
-		} else {
-			cv = g.setSidecarContainerVersionsV0_4()
-		}
+	if g.kubeVersion.GreaterThan(k8sVer1_13) || g.kubeVersion.Equal(k8sVer1_13) {
+		cv = g.getDefaultConfigV1_0()
```

Comment:
so the assumption is PX version is always >= 2.1 since this manifest support is only for 2.6 ?

Reply:
Yes the assumption is that px version will always be > 2.1 because we started supporting operator since 2.1.2. Nothing to do with that manifest is only supported since 2.6. Starting 2.6 we will use new manifest, but for older versions we will still use the older manifest (unless we backport the new manifest on older installer branches, which we are not doing most likely).